resources/assets/coffee/react/beatmapset-page/scoreboard.coffee

```diff
@@ -15,100 +15,81 @@
 # You should have received a copy of the GNU Affero General Public License
 # along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
 ###
-{div, span, p} = React.DOM
+{div, p} = React.DOM
 el = React.createElement
 
-bn = 'beatmapset-scoreboard'
+class BeatmapsetPage.Scoreboard extends React.Component
+  constructor: (props) ->
+    super props
 
-BeatmapsetPage.Scoreboard = React.createClass
-  getInitialState: ->
-    loading: false
+    @state =
+      loading: false
 
+  setLoading: (_e, isLoading) =>
+    @setState loading: isLoading
 
   componentDidMount: ->
     $.subscribe 'beatmapset:scoreboard:loading.beatmapsetPageScoreboard', @setLoading
 
-
   componentWillUnmount: ->
     $.unsubscribe '.beatmapsetPageScoreboard'
 
-
   render: ->
-    scoreboards = ['global', 'country', 'friend']
-    className = "#{bn}__main"
-    className += " #{bn}__main--loading" if @state.loading
-
-    div
-      className: "page-extra #{bn}"
-      el BeatmapsetPage.ExtraHeader, name: 'scoreboard'
-
-      if currentUser.id?
-        div null,
-          div
-            className: "#{bn}__tabs"
-            scoreboards.map (m) =>
-              el BeatmapsetPage.ScoreboardTab,
-                key: m
-                scoreboard: m
-                currentScoreboard: @props.currentScoreboard
-
-          div className: "#{bn}__line"
-
-      div className: className,
+    userScoreFound = false
+
+    mods = ['EZ', 'NF', 'HT', 'HR', 'SD', 'PF', 'DT', 'NC', 'HD', 'FL']
+    if @props.playmode == 'mania' && @props.convert == true
```

Comment:
pretty sure both converts and non-converts can use same set of mods (by selecting mania mode explicitly).